File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/SubtextRule.java
package xyz.wingio.plugins.morehighlight;

import android.content.Context;

import com.discord.simpleast.core.parser.ParseSpec;
import com.discord.simpleast.core.parser.Parser;
import com.discord.simpleast.core.node.Node;
import com.discord.simpleast.core.parser.Rule;
import com.discord.utilities.textprocessing.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
public final class SubtextRule extends Rule.BlockRule<MessageRenderContext, SubtextNode<MessageRenderContext>, MessageParseState> {
    public SubtextRule() {
        super(Pattern.compile("^\\s*(-#)\\s+(.+)(?=\\n|$)"));
    }

    @Override
    public ParseSpec<MessageRenderContext, MessageParseState> parse(Matcher matcher, Parser<MessageRenderContext, ? super SubtextNode<MessageRenderContext>, MessageParseState> parser, MessageParseState s) {
        String content = matcher.group(2);
        
        SubtextNode SubtextNode = new SubtextNode(content);
        return new ParseSpec<>(SubtextNode, s);
    }
}

File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/rule/SubtextRule.kt
package xyz.wingio.plugins.morehighlight.rule

import android.content.Context
import com.discord.simpleast.core.parser.ParseSpec
import com.discord.simpleast.core.parser.Parser
import com.discord.simpleast.core.parser.Rule.BlockRule
import com.discord.utilities.textprocessing.MessageParseState
import com.discord.utilities.textprocessing.MessageRenderContext
import xyz.wingio.plugins.morehighlight.node.BulletPointNode
import xyz.wingio.plugins.morehighlight.node.SubtextNode
import java.util.regex.Matcher
import java.util.regex.Pattern

class SubtextRule(
    private val context: Context
): BlockRule<MessageRenderContext, SubtextNode<MessageRenderContext?>, MessageParseState>(
    Pattern.compile("^\\s*(-#)\\s+(.+)(?=\\n|$)")
) {

    override fun parse(
        matcher: Matcher,
        parser: Parser<MessageRenderContext?, in SubtextNode<MessageRenderContext?>?, MessageParseState?>,
        s: MessageParseState
    ): ParseSpec<MessageRenderContext?, MessageParseState?> {
        return ParseSpec(SubtextNode(context), s, matcher.start(2), matcher.end(2))
    }

}
